modules/tinymce/src/plugins/table/main/ts/selection/SelectionTargets.ts

```diff
@@ -62,18 +62,25 @@ export const getSelectionTargets = (editor: Editor, selections: Selections): Sel
 
   const isCaption = SugarNode.isTag('caption');
   const isDisabledForSelection = (key: keyof ExtractedSelectionDetails) => selectionDetails.forall((details) => !details[key]);
+  const getStart = () => TableSelection.getSelectionStartCellOrCaption(Util.getSelectionStart(editor), Util.getIsRoot(editor));
+  const getEnd = () => TableSelection.getSelectionEndCellOrCaption(Util.getSelectionEnd(editor), Util.getIsRoot(editor));
+
+  const findTargets = (): Optional<RunOperation.CombinedTargets> =>
```

Comment:
Could this be simpler with `Optionals.lift2`?

Something like

```typescript
const start = getStart();
const end = getEnd().bind(TableLookup.table);

return Optionals.lift2(start, end, (startCellOrCaption, endTable) => {
 ...
});
```

That many nested binds looks like an anti pattern to me

Reply:
I did try to do this fwiw, but because we need to return an `Optional` inside we'd end up with the type being `Optional<Optional<RunOperation.CombinedTargets>>` so it still ends up fairly messy to handle that.

Reply:
`.bind(Fun.identity)` once at the end might not be a bad trade-off

Reply:
There's `Optionals.flatten` for that instead, but it's still not pretty.

Reply:
This feels cleaner to me — it separates the "is this in the same table as the end of the selection" out from the other logic with `.filter`
```typescript
  const findTargets = (): Optional<RunOperation.CombinedTargets> =>
    getStart().filter((startCellOrCaption) => {
      const startTableOpt = TableLookup.table(startCellOrCaption);
      const endTableOpt = getEnd().bind(TableLookup.table);
      return Optionals.lift2(startTableOpt, endTableOpt, Compare.eq).getOr(false)
    }).map((startCellOrCaption) => {
      if (isCaption(startCellOrCaption)) {
        return TableTargets.noMenu(startCellOrCaption);
      } else {
        // We know this is okay, because it has to be Some for the previous filter to pass
        const startTable = TableLookup.table(startCellOrCaption).getOrDie();
        return TableTargets.forMenu(selections, startTable, startCellOrCaption);
      }
    });
```

Reply:
Then we're looking up the table twice, and we do the `getOrDie`. That there needs to be a comment saying it's fine speaks against it being an improvement to me.

Reply:
Fair enough, let's ship it as is then